code/go/0chain.net/chaincore/block/entity_test.go

```diff
@@ -0,0 +1,3645 @@
+				if len(b.PrevBlockVerificationTickets) == 0 {
+					b.PrevBlockVerificationTickets = prevB.GetVerificationTickets()
+				}
+
+				return b
+			}(),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			b := &Block{
+				UnverifiedBlockBody:   tt.fields.UnverifiedBlockBody,
+				VerificationTickets:   tt.fields.VerificationTickets,
+				HashIDField:           tt.fields.HashIDField,
+				Signature:             tt.fields.Signature,
+				ChainID:               tt.fields.ChainID,
+				ChainWeight:           tt.fields.ChainWeight,
+				RoundRank:             tt.fields.RoundRank,
+				PrevBlock:             tt.fields.PrevBlock,
+				TxnsMap:               tt.fields.TxnsMap,
+				ClientState:           tt.fields.ClientState,
+				stateStatus:           tt.fields.stateStatus,
+				blockState:            tt.fields.blockState,
+				isNotarized:           tt.fields.isNotarized,
+				ticketsMutex:          tt.fields.ticketsMutex,
+				verificationStatus:    tt.fields.verificationStatus,
+				RunningTxnCount:       tt.fields.RunningTxnCount,
+				UniqueBlockExtensions: tt.fields.UniqueBlockExtensions,
+				MagicBlock:            tt.fields.MagicBlock,
+			}
+
+			b.SetPreviousBlock(tt.args.prevBlock)
+
+			// setting mutexes and states to nil because they are not comparable
+			nilBlocksMutexes(tt.want)
+			nilBlocksMutexes(b)
+
+			if !assert.Equal(t, tt.want, b) {
+				t.Errorf("SetPreviousBlock() got = %v, want = %v", b, tt.want)
+			}
+		})
+	}
+}
+
+func TestBlock_SetStateDB_Debug_True(t *testing.T) {
+	t.Skip("need to protect debug global variable against parallel access")
+
+	t.Parallel()
+
+	type fields struct {
+		UnverifiedBlockBody   UnverifiedBlockBody
+		VerificationTickets   []*VerificationTicket
+		HashIDField           datastore.HashIDField
+		Signature             string
+		ChainID               datastore.Key
+		ChainWeight           float64
+		RoundRank             int
+		PrevBlock             *Block
+		TxnsMap               map[string]bool
+		ClientState           util.MerklePatriciaTrieI
+		stateStatus           int8
+		blockState            int8
+		isNotarized           bool
+		ticketsMutex          *sync.RWMutex
+		verificationStatus    int
+		RunningTxnCount       int64
+		UniqueBlockExtensions map[string]bool
+		MagicBlock            *MagicBlock
+	}
+	type args struct {
+		prevBlock *Block
+	}
+	tests := []struct {
+		name      string
+		fields    fields
+		args      args
+		want      *Block
+		wantPanic bool
+	}{
+		{
+			name:      "Debug_PANIC",
+			wantPanic: true,
+		},
+		// duplicating tests to expose race errors
+		{
+			name:      "Debug_PANIC",
+			wantPanic: true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			defer func() {
+				got := recover()
+				if (got != nil) != tt.wantPanic {
+					t.Errorf("SetStateDB() want panic  = %v, but got = %v", tt.wantPanic, got)
+				}
+			}()
+
+			state.SetDebugLevel(1)
```

Comment:
Move this out of the `for` loop. Set once for all tests, as `state.SetDebugLevel` called once at the start of execution.

Reply:
Moved and fixed.